e2e/aks_model.go

```diff
@@ -179,8 +186,19 @@ func addPrivateEndpointForACR(ctx context.Context, t *testing.T, nodeResourceGro
 	return nil
 }
 
-func createPrivateEndpoint(ctx context.Context, t *testing.T, nodeResourceGroup string, vnet VNet) (armnetwork.PrivateEndpointsClientCreateOrUpdateResponse, error) {
-	endpointName := "PE-for-ABE2ETests"
+func privateEndpointExists(ctx context.Context, t *testing.T, nodeResourceGroup, privateEndpointName string) (bool, error) {
+	existingPE, err := config.Azure.PrivateEndpointClient.Get(ctx, nodeResourceGroup, privateEndpointName, nil)
+	if err == nil && existingPE.ID != nil {
+		t.Logf("Private Endpoint already exists with ID: %s\n", *existingPE.ID)
+		return true, nil
+	}
+	if err != nil && !strings.Contains(err.Error(), "ResourceNotFound") {
```

Comment:
right now we're silencing the error in cases where it doesn't match this pattern, we should probably either always return it or at least always log it in any case if it's non-nil